src/test/interface/pathfinder_test.cpp

```diff
@@ -0,0 +1,114 @@
+  std::vector<std::string> test_arg_diags;
+  std::vector<std::string> test_result_csv;
+  std::vector<std::string> test_result_single;
+  std::vector<std::string> test_result_diags;
+};
+
+TEST_F(CmdStan, pathfinder_good) {
+  std::stringstream ss;
+  ss << convert_model_path(multi_normal_model)
+     << " output refresh=0 file=" << convert_model_path(test_arg_output)
+     << " method=pathfinder num_psis_draws=40";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+
+  std::fstream result_stream(convert_model_path(test_result_csv));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  EXPECT_EQ(1, count_matches("# Elapsed Time:", output));
+  EXPECT_EQ(1, count_matches(" seconds (Pathfinders)", output));
+  EXPECT_EQ(1, count_matches(" seconds (PSIS)", output));
+  EXPECT_EQ(1, count_matches(" seconds (Total)", output));
+  EXPECT_EQ(1, count_matches(" seconds (Total)", output));
+  EXPECT_EQ(1, count_matches("num_psis_draws = 40", output));
+
+  result_sstream.str(std::string());
+  std::fstream single_stream(convert_model_path(test_result_single));
+  result_sstream << single_stream.rdbuf();
+  single_stream.close();
+  output = result_sstream.str();
+  EXPECT_EQ(1, count_matches("# Elapsed Time:", output));
+  EXPECT_EQ(1, count_matches(" seconds (Pathfinder)", output));
+}
+
+TEST_F(CmdStan, pathfinder_single_good) {
+  std::stringstream ss;
+  ss << convert_model_path(multi_normal_model)
+     << " output refresh=0 file=" << convert_model_path(test_arg_output)
+     << " method=pathfinder"
+     << " num_paths=1";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+
+  std::vector<std::string> test_result_1path = {"test", "tmp_pf.csv"};
+  std::fstream result_stream(convert_model_path(test_result_1path));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  EXPECT_EQ(1, count_matches("# Elapsed Time:", output));
+  EXPECT_EQ(1, count_matches(" seconds (Pathfinder)", output));
+}
+
+TEST_F(CmdStan, pathfinder_multi_good) {
+  std::stringstream ss;
+  ss << convert_model_path(multi_normal_model)
+     << " output refresh=0 file=" << convert_model_path(test_arg_output)
+     << " method=pathfinder"
+     << " num_paths=8";
+  run_command_output out = run_command(ss.str());
+  ASSERT_FALSE(out.hasError);
+  std::vector<std::string> test_result_8path = {"test", "tmp_pf_8.csv"};
+  std::fstream result_stream(convert_model_path(test_result_8path));
+  std::stringstream result_sstream;
+  result_sstream << result_stream.rdbuf();
+  result_stream.close();
+  std::string output = result_sstream.str();
+  EXPECT_EQ(1, count_matches("# Elapsed Time:", output));
+  EXPECT_EQ(1, count_matches(" seconds (Pathfinder)", output));
+}
+
+TEST_F(CmdStan, pathfinder_diagnostic_json) {
```

Comment:
Can you also add a test for multi-path for both kinds of diagnostic file?

Reply:
there's only one kind of diagnostic file.